packages/driver/src/cy/listeners.js

```diff
@@ -0,0 +1,107 @@
+const _ = require('lodash')
+
+const HISTORY_ATTRS = 'pushState replaceState'.split(' ')
+
+let events = []
+let listenersAdded = null
+
+const removeAllListeners = () => {
+  listenersAdded = false
+
+  for (let e of events) {
+    const [win, event, cb] = e
+
+    win.removeEventListener(event, cb)
+  }
+
+  // reset all the events
+  events = []
+
+  return null
+}
+
+const addListener = (win, event, cb) => {
+  events.push([win, event, cb])
+
+  win.addEventListener(event, cb)
+}
+
+const eventHasReturnValue = (e) => {
+  const val = e.returnValue
+
+  // return false if val is an empty string
+  // of if its undinefed
+  if (val === '' || _.isUndefined(val)) {
+    return false
+  }
+
+  // else return true
+  return true
+}
+
+module.exports = {
+  bindTo (contentWindow, callbacks = {}) {
+    if (listenersAdded) {
+      return
+    }
+
+    removeAllListeners()
+
+    listenersAdded = true
+
+    // set onerror global handler
+    contentWindow.onerror = callbacks.onError
+
+    addListener(contentWindow, 'beforeunload', (e) => {
+      // bail if we've canceled this event (from another source)
+      // or we've set a returnValue on the original event
+      if (e.defaultPrevented || eventHasReturnValue(e)) {
+        return
+      }
+
+      callbacks.onBeforeUnload(e)
+    })
+
+    addListener(contentWindow, 'unload', (e) => {
+      // when we unload we need to remove all of the event listeners
+      removeAllListeners()
+
+      // else we know to proceed onwards!
+      callbacks.onUnload(e)
+    })
+
+    addListener(contentWindow, 'hashchange', (e) => {
+      callbacks.onNavigation('hashchange', e)
+    })
+
+    for (let attr of HISTORY_ATTRS) {
+      const orig = contentWindow.history?.[attr]
+
+      if (!orig) {
+        continue
```

Comment:
this should be `return`.

Previous CS was:

```coffee
return if not (orig = contentWindow.history?[attr])
```


```suggestion
        return
```

Reply:
It was `return` because it defined a function and called it there. `return` was used to stop the code and continue the loop. 

In JS, that function became unnecessary. So, I removed it and changed `return` to `continue` to stop the code and continue on to the next item.

Reply:
Elaboration:

The code below is the JS code converted by decaffeinate. As you can see, there is an unnecessary function definition and call. 

```js
for (let attr of HISTORY_ATTRS) {
  (function(attr) {
    let orig;
    if (!(orig = contentWindow.history != null ? contentWindow.history[attr] : undefined)) { return; }

    return contentWindow.history[attr] = function(...args) {
      orig.apply(this, args);

      return callbacks.onNavigation(attr, args);
    };
  })(attr);
}
```

I removed it and `return` to `continue` was necessary change to make it work.